server/src/main/java/keywhiz/inject/InjectorFactory.java

```diff
@@ -0,0 +1,24 @@
+package keywhiz.inject;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import io.dropwizard.db.ManagedDataSource;
+import io.dropwizard.setup.Environment;
+import keywhiz.KeywhizConfig;
+import keywhiz.ServiceModule;
+
+public final class InjectorFactory {
+  private InjectorFactory() {}
+
+  public static Injector createInjector(KeywhizConfig config, Environment environment) {
+    return Guice.createInjector(
+        new ServiceDataSourceModule(),
+        new ServiceModule(config, environment));
+  }
+
+  public static Injector createInjector(KeywhizConfig config, Environment environment, ManagedDataSource dataSource) {
+    return Guice.createInjector(
+        new DataSourceModule(dataSource, dataSource),
+        new ServiceModule(config, environment));
+  }
+}
```

Comment:
Our handy factory class is used by the service ("server" command), other commands, and by tests. This ensures that our construction is unified for all use cases.

As mentioned earlier, we could get rid of the overload by doing a little more work inside the KeywhizService doRun() method as far as assembling health checks, eliminating the need for ServiceDataSourceModule.